test/e2e/framework/pod/resize.go

```diff
@@ -350,88 +320,83 @@ func VerifyPodContainersCgroupValues(ctx context.Context, f *framework.Framework
 			}
 			if expectedMemLimitString != "0" {
-				err := verifyCgroupValue(ci.Name, cgroupMemLimit, expectedMemLimitString)
-				if err != nil {
-					return err
-				}
-			}
-			err := verifyCgroupValue(ci.Name, cgroupCPULimit, expectedCPULimitString)
-			if err != nil {
-				return err
-			}
-			err = verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(expectedCPUShares, 10))
-			if err != nil {
-				return err
+				errs = append(errs, verifyCgroupValue(ci.Name, cgroupMemLimit, expectedMemLimitString))
 			}
+			errs = append(errs, verifyCgroupValue(ci.Name, cgroupCPULimit, expectedCPULimitString))
+			errs = append(errs, verifyCgroupValue(ci.Name, cgroupCPURequest, strconv.FormatInt(expectedCPUShares, 10)))
 		}
 	}
-	return nil
+	return utilerrors.NewAggregate(errs)
 }
 
-func waitForContainerRestart(ctx context.Context, podClient *PodClient, pod *v1.Pod, expectedContainers []ResizableContainerInfo, initialContainers []ResizableContainerInfo, isRollback bool) error {
+func verifyContainerRestarts(pod *v1.Pod, expectedContainers []ResizableContainerInfo) error {
 	ginkgo.GinkgoHelper()
-	var restartContainersExpected []string
 
-	restartContainers := expectedContainers
-	// if we're rolling back, extract restart counts from test case "expected" containers
-	if isRollback {
-		restartContainers = initialContainers
+	expectContainerRestarts := map[string]int32{}
+	for _, ci := range expectedContainers {
+		expectContainerRestarts[ci.Name] = ci.RestartCount
 	}
 
-	for _, ci := range restartContainers {
-		if ci.RestartCount > 0 {
-			restartContainersExpected = append(restartContainersExpected, ci.Name)
+	errs := []error{}
+	for _, cs := range pod.Status.ContainerStatuses {
+		expectedRestarts := expectContainerRestarts[cs.Name]
+		if cs.RestartCount != expectedRestarts {
+			errs = append(errs, fmt.Errorf("unexpected number of restarts for container %s: got %d, want %d", cs.Name, cs.RestartCount, expectedRestarts))
 		}
 	}
-	if len(restartContainersExpected) == 0 {
-		return nil
-	}
+	return utilerrors.NewAggregate(errs)
+}
 
-	pod, err := podClient.Get(ctx, pod.Name, metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-	restartedContainersCount := 0
-	for _, cName := range restartContainersExpected {
-		cs, _ := podutil.GetContainerStatus(pod.Status.ContainerStatuses, cName)
-		if cs.RestartCount < 1 {
-			break
+func WaitForPodResizeActuation(ctx context.Context, f *framework.Framework, podClient *PodClient, pod *v1.Pod) *v1.Pod {
+	ginkgo.GinkgoHelper()
+	// Wait for resize to complete.
+	framework.ExpectNoError(WaitForPodCondition(ctx, f.ClientSet, pod.Namespace, pod.Name, "resize status cleared", f.Timeouts.PodStart,
+		func(pod *v1.Pod) (bool, error) {
+			if pod.Status.Resize == v1.PodResizeStatusInfeasible {
+				// This is a terminal resize state
+				return false, fmt.Errorf("resize is infeasible")
+			}
+			return pod.Status.Resize == "", nil
+		}), "pod should finish resizing")
```

Comment:
It seems that resize status is updated to `Proposed` only when requests are resized:
https://github.com/kubernetes/kubernetes/blob/3f5d0ee2cf402d10233fc403750a9784d1542059/pkg/api/pod/util.go#L1304-L1306

Test cases to resize only resource limits may fail by seeing empty resize status before kubelet sets `InProgress`.

Reply:
Awesome, thanks for catching this! I've been butting my head against this test failure for a day! Fix coming soon...